File: _CodeTopics/LeetCode_contest/weekly/weekly2021/232-[smallweek]/232_1.py
class Solution(object):
    def areAlmostEqual(self, s1, s2):
        """
        :type s1: str
        :type s2: str
        :rtype: bool
        """
        
        ind = []
        for i in range(len(s1)):
            if s1[i] != s2[i]:
                ind.append(i)
                if len(ind) > 2:
                    return False
        if not ind or (len(ind) == 2 and s1[ind[0]] == s2[ind[1]] and s1[ind[1]] == s2[ind[0]]):
            return True
        return False
    
"""
https://leetcode-cn.com/submissions/detail/154892646/

113 / 113 个通过测试用例
状态：通过
执行用时: 20 ms
内存消耗: 13.1 MB
"""

File: _CodeTopics/LeetCode/1601-1800/001790--weekly232_1/tran/232_1.cpp
class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {

        // 第 232 场周赛第一题
        // https://leetcode.cn/submissions/detail/154892646/

        vector<int> ind = {};
        for (int i = 0; i < s1.size(); ++i) {
            if (s1[i] != s2[i]) {
                ind.push_back(i);
                if (ind.size() > 2) {
                    return false;
                }
            }
        }
        if (ind.empty() || (ind.size() == 2 && s1[ind[0]] == s2[ind[1]] && s1[ind[1]] == s2[ind[0]])) {
            return true;
        }
        return false;
    }
};

/*
https://leetcode.cn/submissions/detail/371773316/

执行用时：
4 ms
, 在所有 C++ 提交中击败了
17.66%
的用户
内存消耗：
6.1 MB
, 在所有 C++ 提交中击败了
24.69%
的用户
通过测试用例：
131 / 131
*/
